### ISSUE TITLE
Corrupted file

### ISSUE THREAD

**Issue by 9696-ac** [LMMS/lmms#4533]
Hello, 
So I've been working on a project for hours.

I was working on this file, then LMMS froze for a second and a shortcut (ctrl+drag) didnt work, so I saved, closed it and restarted it to find that the file is now completely empty with no tracks (so I reopened it therefore, the recover.mmp won't work). 
I went to check my project files hoping to restore its backup .bak file, but both the project and backup file is now 1.92KB 
So theyre corrupted, however there were files in the Temp folder corresponding to times I opened and modified the file.

Is there anything I can do to repair this file... 
Is it possible to restore a file's contents with content from the Temp Folder? (other than changing the file type, as this seems to work for Office documents, but not for this) 
Or use a temporary file to create a restore point for the file? 

I really dont want to lose what I've done (many many hours) and theres no way I can remember exactly what I did to recreate it... 
Thankyou for your time.

**Comment by 9696-ac** [comment]
Hello, 
Thankyou for your reply. 
I am using LMMS 1.2.0 on Windows 10. 
Version: 1803 
OS build: 17134.165 
Here is a link to the temp files and the mentioned corrupted file. 
https://drive.google.com/open?id=1rUXho_GQGr-XBKbJaED0B1XlTEjZgUlM
I apologize ahead of time if I have made misconceptions about these temp files. 

Thankyou for your time.

**Comment by musikBear** [comment]
@9696-ac Sadly the files are only zynadsubfx data files as zonk thought. There are no project data. The corrupted file however, is not corrupted, it opens perfectly fine, but has no *content* at all. 
Have you scanned for any files with either *.BAK* as extention, or named *recover.mmp*
@zonkmachine imo having both the *backup-to-BAK* and recover within timeslot 10 mins, should not only be default enabled, but should be added to 1.20, before it goes RC7. What do you think?

**Comment by zonkmachine** [comment]
> @zonkmachine imo having both the backup-to-BAK and recover within timeslot 10 mins, should not only be default enabled, but should be added to 1.20, before it goes RC7. What do you think?

They already are on by default.

**Comment by 9696-ac** [comment]
@musikBear 
I have and as mentioned previously, 
The .bak file was at 2KB , the same as the original file. 
The recover.mmp file was replaced by me... because I reopened the file without knowing it the file was 'empty'. 

Why would a file completely delete all its content if it wasnt corrupted though... 
Or was it because i saved after it had an issue?

**Comment by musikBear** [comment]
@9696-ac  >was it because i saved after it had an issue?

That is properly the reason, but i cant understand where your BAK file went, that should have been the version *before* your failed save, and since @zonkmachine says recover is on already as default, then you should also have had a recover-file. Did you get the recover-file screen?
Eg: https://cloud.githubusercontent.com/assets/189962/18536592/8544a7b0-7b52-11e6-9379-d7e97a4dc751.png 
If you did, what option did you choose?

**Comment by 9696-ac** [comment]
@musikBear 
That screen only appears if you close it without saving right? (->last session did not end properly) 
However, I saved the file right before I closed it so, no project was closed un-saved. Which shouldnt trigger that pop-up. 
If my knowledge is correct...

**Comment by musikBear** [comment]
@9696-ac > However, I saved the file right before I closed it so, no project was closed un-saved

No this is where you are mistaken. You wrote
>I was working on this file, then LMMS froze for a second and a shortcut (ctrl+drag) didnt work, so I saved.

No you _thought_ you saved!
Infact LMMS was frozen, but that does ofcause not prevent you to press ctrl+s on your keyboard. I doubt LMMS responded(?) You should get at status info in botton left. -Ofcause you cant remember if you got that now.
I am rather sure, that you never saved the edited project, however that cant explain why the empty un-corrupted BAK file existed, because a BAK is only made when the project is manually saved.. *That* is puzzling. You ought to have had one file with content, **unless** you actually saved once, before you made a single change in the project, *then* both mmpz and BAK would be empty -Does that ring a bell? 

(Its rather important that this is understood, because a fail-save, like the one in 4514, but different -is a real bad bug -Its remind me about consensus on having an alert coloured fail-save warning (like red) ..I cant find the issue, though. Remember that conversation @zonkmachine ?

**Comment by 9696-ac** [comment]
@musikBear 
Ok, so from what I've gathered, and what I thought I knew. 

The .bak file saves the last time I manually saved.  
And the only reasons why the file would be empty is if, my last manual save was while the project was empty (which is not the case, I save almost every change make, almost too often and I worked on this for up to 6 hours) or, a bug? 
 About the status messsge, I remembered one coming up, but not responding. Of course i could remember wrong
